[libtorch] [xnnpack] [qnnpack] [cpuinfo][fbgemm]

EvilMcStevil opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Commit:
[opencv4] add option to disable tracing

CV_TRACE can cause memory leaks https://github.com/opencv/opencv/issues/19727

Commit:
Update opencv4 versions database to 4.11.0 port version -3

Commit:
Update fbgemm to 1.0.0

Commit:
update fbgemm version database

Commit:
removed failure mark for fbgemm 64-android which passed ci

Commit:
updated cpuinfo for to latest version

Commit:
added baseline

Commit:
updated qnnpack to build again with new cpuinfo

Commit:
updated to simplify patch

Commit:
removed bad spacing in diff to simplify again

Commit:
update qnnpack baseline

Commit:
added xnnpack change

Commit:
add xnnpack baseline

Commit:
updated baseline

Commit:
initial patches for libtorch

Commit:
added clog dep to cpuinfo  port file

Commit:
updated qnnpack to use clog compiled from cpuinfo

Commit:
format manifests

Commit:
updated version baselines

Commit:
added python to xnnpack build requirements

Commit:
added kliedai as dependecy to arm64

Commit:
diabled cpu info clog tests

Commit:
fixed spelling mistake on kleidiai dependency

Commit:
Removed notrace feature and set CV_TRACE to be off

Commit:
updated opencv4 baseline

Commit:
some cmake deps fixed

Commit:
updated nvidia cutlass and added utils tools as needed by libtorch 2.7.1

Commit:
Added baseline

Commit:
removed unneeded patch after building

Commit:
fixed baseline

Commit:
Merge branch 'nvidia-cutlass'

Commit:
updated libtorch

Commit:
added nvtx3

Commit:
updated libtorch port to compile a shared object

Commit:
moved plibtorch patches back

Commit:
Added vcpkg_find_cuda to portfile

Commit:
Added Missing Feature Options for nvidia compiler

Commit:
Merge branch 'nvidia-cutlass' into torch271

# Conflicts:
#	ports/nvidia-cutlass/portfile.cmake
#	versions/n-/nvidia-cutlass.json

Commit:
added pybind11

Commit:
improved and cmake config to xnnpack exports

Commit:
improved and cmake config to xnnpack exports

Commit:
updated build trees

Commit:
updated libtorch for windows build

Commit:
correcxt case of dependency

EvilMcStevil committed:
Merge branch 'microsoft:master' into xnnpack-qnnpack-cpuinfo

dg0yt commented on ports/qnnpack/use-packages.patch:
```diff
@@ -105,26 +100,27 @@ index a5ddc49..05e1279 100644
++
  
  # ---[ Configure cpuinfo
 -IF(NOT TARGET cpuinfo)
 +IF(FALSE)
```
I would prefer this here
```suggestion
+find_package(cpuinfo CONFIG REQUIRED)
+add_library(cpuinfo ALIAS cpuinfo::cpuinfo)
 IF(NOT TARGET cpuinfo)
```
so that it is one local, minimal change.
And similar for clog.
And there has to be `find_dependency(cpuinfo REQUIRED)` in the exported config.

dg0yt commented on ports/xnnpack/fix-build.patch:
```diff
@@ -1,94 +1,72 @@
-@@ -881,12 +884,12 @@ IF(NOT TARGET fxdiv)
+ 
+ # ---[ Configure cpuinfo
++find_package(cpuinfo CONFIG REQUIRED)
++add_library(cpuinfo ALIAS cpuinfo::cpuinfo)
++
+ IF(NOT TARGET cpuinfo)
```
... like it is already done here.

dg0yt commented on versions/q-/qnnpack.json:
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "c9db1bdd7c82e0bbe3a1fa3599263f70781e8a61",
+      "version-date": "2021-02-26",
+      "port-version": 5
+    },
+    {
+      "git-tree": "6cfef60767e1bc650df581901e4d14e132d5fdea",
+      "version-date": "2021-02-26",
+      "port-version": 4
+    },
```
For each port, you can only add one port version (`4`) in one PR.

Commit:
Merge remote-tracking branch 'github.com/master' into torch271

# Conflicts:
#	ports/fbgemm/fix-cmakelists.patch
#	versions/f-/fbgemm.json

Commit:
Merge branch 'torch271' into xnnpack-qnnpack-cpuinfo

Commit:
added libtorch baselines

Commit:
added ability to find fxlib as a cmake package, to simplify dependent package configurations (xnnpack / libtorch / pthreadpool etc)

Commit:
added using fxdiv as package

Commit:
updated qnnpack

Commit:
updated

Commit:
added dependecy to pthreadspool config

Commit:
added cmake package to psimd

Commit:
added cmake package to psimd

Commit:
added cmake config to fp16

Commit:
added cmake config to fp16

Commit:
qnnpack now uses config files

Commit:
xnnpack config fixes

Commit:
add more config deps to lib torch

Commit:
[fxdiv] Add fxdiv-config.cmake files

Commit:
Added baseline

Commit:
updated psimd baseline

Commit:
added fp16 new baseline

Commit:
Merge branch 'fxdiv-config' into xnnpack-qnnpack-cpuinfo

Commit:
Merge branch 'psimd' into xnnpack-qnnpack-cpuinfo

Commit:
Merge remote-tracking branch 'origin/xnntemp' into xnnpack-qnnpack-cpuinfo

Commit:
new baselines

Commit:
Merge remote-tracking branch 'github.com/master' into xnnpack-qnnpack-cpuinfo

# Conflicts:
#	versions/p-/pthreadpool.json

Commit:
removed leveldb option as not in code anywhere

Commit:
added nnpack cmake bindings

Commit:
added nnpack cmake bindings

BillyONeal commented:
This is depends:vm-update to ensure nvtx (and all other CUDA libraries) are included in the Linux containers. Moreover, CUDA should be upgraded to 12.9 (I missed this in #46371 )

Commit:
Merge branch 'nnpack' into xnnpack-qnnpack-cpuinfo

Commit:
update torch so nnpack dep compiles

Commit:
made some gflags and glog optional depedencies

Commit:
undone rername of nnpack project

Commit:
made nnpack bindings unofficial

EvilMcStevil committed:
Update ports/nnpack/portfile.cmake

Co-authored-by: Kai Pastor <dg0yt@darc.de>

Commit:
Merge remote-tracking branch 'origin/nnpack' into xnnpack-qnnpack-cpuinfo

Commit:
updated nnpack baseline

Commit:
Merge branch 'nnpack' into xnnpack-qnnpack-cpuinfo

Commit:
added tensor pipe to dist build

Commit:
updated packages to use unofficial names for cmake, and fixed a dependecy bug on windows builds

Commit:
added missing unofficial to find_dependency

Commit:
made cmake bindings unofficial

Commit:
Merge branch 'fxdiv-pthreadpool' into nnpack

Commit:
Merge branch 'fp16-psimd-unnofficial' into nnpack

Commit:
updated nnpack to use unofficial bindings

Commit:
Merge branch 'nnpack' into xnnpack-qnnpack-cpuinfo

Commit:
made xnn pack look for unofficial fxdiv

Commit:
updated libtroch xnnpack to use unofficial ports

luncliff committed:
[libtorch] patch for CUDA build in Windows

# Conflicts:
#	ports/libtorch/fix-cmake.patch
#	ports/libtorch/portfile.cmake
#	versions/l-/libtorch.json

EvilMcStevil committed:
Merge branch 'microsoft:master' into xnnpack-qnnpack-cpuinfo

EvilMcStevil committed:
Delete ports/nnpack/make-unofficial-bindings.patch

Commit:
qnnpack now uses unofficial packages

Commit:
added feture baseline fail for libtorch vulkan

Commit:
put back cascade

Commit:
linux feature fail

Commit:
updated fbgem cmake bindings to export dependecies and correctly add FGEMM_STATIC to them

Commit:
added fbgemm baseline

Commit:
added patch back in

Commit:
xnnpack debug cmake install fixed

Commit:
updated baselines

Commit:
fixed magma baseline file

Commit:
fixed onnx dependency

Commit:
osx fix

EvilMcStevil committed:
Build output dir fixes

Commit:
updated libtorch binding

EvilMcStevil committed:
added vulkan fix patch back, fixed error in CUDA port

EvilMcStevil committed:
updated version binding

Commit:
Removed vulkan cascade error

Commit:
added vulkan fix

Commit:
marking windows cuda as expected to fail in ci,
[1441/1606] C:\PROGRA~1\NVIDIA~2\CUDA\v12.9\bin\nvcc.exe -forward-unknown-to-host-compiler -DABSL_CONSUME_DLL -DAT_PER_OPERATOR_HEADERS -DCUTLASS_ENABLE_CUBLAS=1 -DCUTLASS_ENABLE_CUDNN=1 -DEXPORT_AOTI_FUNCTIONS -DFMT_HEADER_ONLY=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DNOMINMAX -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx -DPROTOBUF_USE_DLLS -DTORCH_CUDA_BUILD_MAIN_LIB -DTORCH_CUDA_USE_NVTX3 -DUSE_CUDA -DUSE_EXTERNAL_MZCRC -DUSE_MEM_EFF_ATTENTION -DUSE_MIMALLOC -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE=1 -D_UCRT_LEGACY_INFINITY -Dtorch_cuda_EXPORTS -ID:\b\libtorch\x64-windows-release-rel\aten\src -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src -ID:\b\libtorch\x64-windows-release-rel -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\THC -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\cuda -ID:\b\libtorch\x64-windows-release-rel\caffe2\aten\src -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\c10\cuda\..\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\c10\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\torch\csrc\api -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\torch\csrc\api\include -isystem D:\installed\x64-windows-release\include -isystem D:\installed\x64-windows-release\include\eigen3 -isystem "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\include" -isystem D:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\cmake\..\third_party\cudnn_frontend\include -DLIBCUDACXX_ENABLE_SIMPLIFIED_COMPLEX_OPERATIONS -Xcompiler  /Zc:__cplusplus -Xcompiler /w -w -Xcompiler /FS -Xfatbin -compress-all -DONNX_NAMESPACE=onnx --use-local-env -gencode arch=compute_50,code=sm_50 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_89,code=sm_89 -gencode arch=compute_90,code=sm_90 -gencode arch=compute_90a,code=sm_90a -gencode arch=compute_100,code=sm_100 -gencode arch=compute_100a,code=sm_100a -gencode arch=compute_101a,code=sm_101a -gencode arch=compute_120,code=sm_120 -gencode arch=compute_120a,code=sm_120a -Xcudafe --diag_suppress=cc_clobber_ignored,--diag_suppress=field_without_dll_interface,--diag_suppress=base_class_has_different_dll_interface,--diag_suppress=dll_interface_conflict_none_assumed,--diag_suppress=dll_interface_conflict_dllexport_assumed,--diag_suppress=bad_friend_decl --Werror cross-execution-space-call --no-host-device-move-forward --expt-relaxed-constexpr --expt-extended-lambda  -Xcompiler=/wd4819,/wd4503,/wd4190,/wd4244,/wd4251,/wd4275,/wd4522 -Wno-deprecated-gpu-targets --expt-extended-lambda -DCUB_WRAPPED_NAMESPACE=at_cuda_detail -DCUDA_HAS_FP16=1 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -Xcompiler="-O2 -Ob2" -DNDEBUG -Xcompiler /MD -std=c++17 -Xcompiler=-MD -MD -MT caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj -MF caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj.d -x cu -c D:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\native\cuda\SegmentReduce.cu -o caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj -Xcompiler=-Fdcaffe2\CMakeFiles\torch_cuda.dir\,-FS
FAILED: caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/SegmentReduce.cu.obj
C:\PROGRA~1\NVIDIA~2\CUDA\v12.9\bin\nvcc.exe -forward-unknown-to-host-compiler -DABSL_CONSUME_DLL -DAT_PER_OPERATOR_HEADERS -DCUTLASS_ENABLE_CUBLAS=1 -DCUTLASS_ENABLE_CUDNN=1 -DEXPORT_AOTI_FUNCTIONS -DFMT_HEADER_ONLY=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DNOMINMAX -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx -DPROTOBUF_USE_DLLS -DTORCH_CUDA_BUILD_MAIN_LIB -DTORCH_CUDA_USE_NVTX3 -DUSE_CUDA -DUSE_EXTERNAL_MZCRC -DUSE_MEM_EFF_ATTENTION -DUSE_MIMALLOC -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE=1 -D_UCRT_LEGACY_INFINITY -Dtorch_cuda_EXPORTS -ID:\b\libtorch\x64-windows-release-rel\aten\src -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src -ID:\b\libtorch\x64-windows-release-rel -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\THC -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\cuda -ID:\b\libtorch\x64-windows-release-rel\caffe2\aten\src -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\c10\cuda\..\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\c10\.. -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\torch\csrc\api -ID:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\torch\csrc\api\include -isystem D:\installed\x64-windows-release\include -isystem D:\installed\x64-windows-release\include\eigen3 -isystem "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\include" -isystem D:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\cmake\..\third_party\cudnn_frontend\include -DLIBCUDACXX_ENABLE_SIMPLIFIED_COMPLEX_OPERATIONS -Xcompiler  /Zc:__cplusplus -Xcompiler /w -w -Xcompiler /FS -Xfatbin -compress-all -DONNX_NAMESPACE=onnx --use-local-env -gencode arch=compute_50,code=sm_50 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_89,code=sm_89 -gencode arch=compute_90,code=sm_90 -gencode arch=compute_90a,code=sm_90a -gencode arch=compute_100,code=sm_100 -gencode arch=compute_100a,code=sm_100a -gencode arch=compute_101a,code=sm_101a -gencode arch=compute_120,code=sm_120 -gencode arch=compute_120a,code=sm_120a -Xcudafe --diag_suppress=cc_clobber_ignored,--diag_suppress=field_without_dll_interface,--diag_suppress=base_class_has_different_dll_interface,--diag_suppress=dll_interface_conflict_none_assumed,--diag_suppress=dll_interface_conflict_dllexport_assumed,--diag_suppress=bad_friend_decl --Werror cross-execution-space-call --no-host-device-move-forward --expt-relaxed-constexpr --expt-extended-lambda  -Xcompiler=/wd4819,/wd4503,/wd4190,/wd4244,/wd4251,/wd4275,/wd4522 -Wno-deprecated-gpu-targets --expt-extended-lambda -DCUB_WRAPPED_NAMESPACE=at_cuda_detail -DCUDA_HAS_FP16=1 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -Xcompiler="-O2 -Ob2" -DNDEBUG -Xcompiler /MD -std=c++17 -Xcompiler=-MD -MD -MT caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj -MF caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj.d -x cu -c D:\b\libtorch\src\v2.7.1-2650bb1bd5.clean\aten\src\ATen\native\cuda\SegmentReduce.cu -o caffe2\CMakeFiles\torch_cuda.dir\__\aten\src\ATen\native\cuda\SegmentReduce.cu.obj -Xcompiler=-Fdcaffe2\CMakeFiles\torch_cuda.dir\,-FS
LLVM ERROR: out of memory

Commit:
made fbgemm only for x64 builds

Commit:
uopdated baseline

Commit:
fixed use fbgemm override

Commit:
added kleidai as dependecy in cmake binding of xnnpack

EvilMcStevil committed:
added fix glog back - needed on osx builds

Commit:
renamed fix build to fix cmake to make comparisons with luncliffs port easier

Commit:
added correct find library to xnnpack for kleidiai

Commit:
removed cuda from default windows build

EvilMcStevil committed:
removed cascade failures for arm64 osx

EvilMcStevil committed:
removed nvtx3 port as not wante in vcpkg see #46694

EvilMcStevil commented:
Also credit to @luncliff as i shamelessly stole the OSX fixes from #46700

EvilMcStevil committed:
made cuda builds expect to complete

BillyONeal commented:
Wait I'm confused; I thought we were waiting for Linux to have nvtx inside but this is green already?

EvilMcStevil commented:
I did have the nvtx3 port in the patch to get past that hurdle for testing, then i removed it, to get ready for the final build, and it passed all the tests,  I did think i saw the system using cuda 12.9 in the builds, but don't have the passing logs to see what it did use. ( i may have seen the windows one) .
Unless having nvtx3 headers available at one time allowed it to work?

BillyONeal commented:
As of #46911 NVTX should be on the machines.

EvilMcStevil committed:
Merge branch 'microsoft:master' into xnnpack-qnnpack-cpuinfo

Commit:
add cuda linkage for tensorpipe

Commit:
updated baselins

Commit:
fixed dist cuda

Commit:
disable dist on osx

EvilMcStevil committed:
Merge branch 'microsoft:master' into xnnpack-qnnpack-cpuinfo
